pkg/asset/installconfig/azure/validation.go

```diff
@@ -89,23 +94,35 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 
 	// Default requirements need to be sufficient to support Control Plane instances.
 	defaultInstanceReq := controlPlaneReq
+	defaultDiskType := ""
+
+	if ic.Platform.Azure.DefaultMachinePlatform != nil {
+		defaultDiskType = ic.Platform.Azure.DefaultMachinePlatform.OSDisk.DiskType
+	}
 
 	if ic.ControlPlane != nil && ic.ControlPlane.Platform.Azure != nil && ic.ControlPlane.Platform.Azure.InstanceType != "" {
```

Comment:
This conditional is still not quite right. The check for `ValidateInstanceType` needs to happen even if the instance type is not set explicitly on the control plane machine. That is because the user could set the disk type explicitly for the control plane machine without setting the instance type.

My suggestion is to do the following.
1. Determine the actual instance type used for the control plane machines and the compute machines in the same manner in which you determine the actual disk type used.
1. Get rid of the validation on the default machine platform.

Reply:
Isn't it possible for the user to set the DefaultMachinePlatform.InstanceType in the install-config file?

Reply:
Yes. And that was covered by https://github.com/openshift/installer/blob/f85551fa5b0e94fb6083bec91e642bfddd3c5e55/pkg/asset/installconfig/azure/validation.go#L100-L102.

Reply:
But that is not sufficient any more, since we need to consider both the instance type and the disk type from the default machine platform.

Reply:
> Get rid of the validation on the default machine platform.
I think I got confused with this point. I understand that I need to figure out the actual instance type used but could to tell me which validation to get rid of here?

Reply:
Get rid of the code that I linked two comments up.

Reply:
> But that is not sufficient any more, since we need to consider both the instance type and the disk type from the default machine platform.

Was thinking since the installer sets default values to both instance type and disk type, if user sets either of these, we could still validate with the installer defaults. This would apply to control-plane, compute and defaultMachinePlatform.